lib/cylc/conditional_simplifier.py

```diff
@@ -108,23 +109,30 @@ def clean_expr( self, nested_list, criteria ):
                isinstance(cleaned[0], list)):
             cleaned = cleaned[0]
 
+        if len(cleaned) == 1:
+            cleaned = cleaned[0]
+
         # Recurse through the nested list and remove criteria.
-        found = None
-        if isinstance(cleaned, str) or len(cleaned)==1:
+        if isinstance(cleaned, str):
             if cleaned == criteria:
                 return ""
             else:
                 return cleaned
+
+        found = None
         for i in range(0, len(cleaned)):
             if isinstance(cleaned[i], list):
                 cleaned[i] = self.clean_expr(cleaned[i], criteria)
             if cleaned[i] == criteria:
                 found = i
+                break
+
         if found is not None:
             if found == 0:
-                return self.clean_expr(cleaned[2], criteria)
-            elif found == 2:
-                return self.clean_expr(cleaned[0], criteria)
```

Comment:
The value of `found` in the example suite of #1626 was 4, so nothing was returned from this method.